spec/requests/apis/unit_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe 'Units API' do
+  context 'GET /api/v1/condos/{id}/units' do
+    it 'successfully' do
+      condo = create :condo
+      first_tower = create :tower, condo:, floor_quantity: 2, units_per_floor: 2
+      second_tower = create :tower, condo:, floor_quantity: 3, units_per_floor: 2
+      first_unit_type = create(:unit_type, description: 'Duplex com varanda', condo:)
+      second_unit_type = create(:unit_type, description: 'Apartamento 1 quarto', condo:)
+
+      first_tower.floors.each do |floor|
```

Comment:
Uma sugestão de melhoria, não quer dizer que o que vocês fizeram está errado, é só uma sugestão que se acharem válido, podem aceitar. 

Vi que a API só retorna as unidades que possuem tipos de unidade associada, uma forma que vc poderiam fazer para retornar todas as unidades com tipos de unidades associada ou não, seria alterar a relação entre o condo e unidade.
Ao inves da relação ocorrer via tipo de unidade, poderia ser feito via torres e andares. 
Assim, vcs conseguiriam ter todas as unidades que já há associação ou não com tipo de unidade.

Como disse, só uma sugestão, se o objetivo for buscar unidades que possuem tipos de unidade associado, então desconsidere.

Reply:
Acredito que o objetivo seja mesmo retornar as unidades com os tipos de unidade associados, até porque ficaria um tanto sem sentido retornar alguma unidade vazia. Mas concordo!

Reply:
É verdade, isso acabou sendo implementado na tua branch né?